rust/src/dns/detect.rs

```diff
@@ -114,9 +115,37 @@ pub unsafe extern "C" fn rs_dns_detect_opcode_free(ptr: *mut c_void) {
     }
 }
 
+/// Perform the DNS rcode match.
+///
+/// 1 will be returned on match, otherwise 0 will be returned.
+#[no_mangle]
+pub extern "C" fn rs_dns_rcode_match(
+    tx: &mut DNSTransaction, detect: &mut DetectUintData<u8>, flags: u8,
+) -> u8 {
+    let header_flags = if flags & Direction::ToServer as u8 != 0 {
+        if let Some(request) = &tx.request {
+            request.header.flags
+        } else {
+            return 0;
+        }
+    } else if let Some(response) = &tx.response {
+        response.header.flags
+    } else {
+        return 0;
+    };
+
+    let rcode = (header_flags & 0xf) as u8;
```

Comment:
As Philippe asked herehttps://github.com/OISF/suricata/pull/10233/files/a122126f2541f3e09c703de68444f39683e7848c#r1464948794 - could this be 23, in the end, as we know from the specifications that it can?

Reply:
I'm not really sure if I understand what you mean.

Reply:
Although this needed to be addressed, I introduced some noise with my question.
But we've discussed this in our weekly call, conclusion was that:
- commit message should be updated
- documentation should indicate that current version covers the RFC specification for rcodes up to 15, and share redmine ticket link as well as related RFC.